Removing code that incorrectly references other tools

michaelpnelson opened the pull request:
This comments out code that references other tools by using invalid tool names. Tool names were once all lowercase but are now camel case, but some references to the old tool names remain. This comments out such code because it is currently not functional. This may help prevent bugs, and by preserving the code we keep its intent should we want to bring back functionality later.

This also comments out nonfunctional buttons in the Search tool. The Identify and Measure buttons do not have implementation code, and the code called by the Directions button results in a console error because of an undefined value.

michaelpnelson committed:
Commenting out code that incorrectly references other tools

This comments out code that references other tools by using invalid tool names. Tool names were once all lowercase but are now camel case, but some references to the old tool names remain. This comments out such code because it is currently not functional. This may help prevent bugs, and by preserving the code we keep its intent should we want to bring back functionality later.

This also comments out nonfunctional buttons in the Search tool. The Identify and Measure buttons do not have implementation code, and the code called by the Directions button results in a console error because of an undefined value.

michaelpnelson commented on src/smk/tool/search/tool-search-location.js:
```diff
@@ -42,26 +42,26 @@ include.module( 'tool-search.tool-search-location-js', [
-            } )
+            //         smk.$tool.directions.activating
+            //             .then( function () {
+            //                 return smk.$tool.directions.startAtCurrentLocation()
```
I spent a while trying to update references to "directions" with the new tools (it was replaced by three - DirectionsOptionsTool, DirectionsRouteTool, and DirectionsWaypointsTool, the last seeming to be the most appropriate) but I wasn't able to get it working.

dgboss commented on src/smk/tool/location/tool-location.js:
```diff
@@ -31,20 +31,20 @@ include.module( 'tool-location', [
             this.setIdentifyHandler = function ( handler ) {
-                if ( !smk.$tool.identify ) return
+                // if ( !smk.$tool.identify ) return
 
```
I'm not a fan of leaving commented out code in the code base, especially since Git will remember it for us. If you choose to go this route, could you please add comments to each section you are commenting out stating why you are doing it? Also, please consider commenting out code at a 'higher level' as well. For example, you might as well comment out the entire this.setIdentifyHandler function itself and comment out any calls to this function. Ditto for this.setDirectionsHandler.

michaelpnelson replied:
Thanks for your thoughts. I'm persuaded to remove the code instead of commenting it out. It's always more fun to remove unused code too :) . My only concern is being able to find this removed code some later day should we want to reinstate functionality ... much git tooling only shows the most recent change in a file/on a line, so it might be harder to see when it was removed. If you have suggestions, let me know. I think I'll update the commit message's first line to be very clear as to what was removed and why, since the first line is what we'll see in git history.

dgboss commented on src/smk/tool/location/tool-location.js:
```diff
@@ -81,15 +81,15 @@ include.module( 'tool-location', [
-                                .then( function () {
-                                    return smk.$tool.directions.addWaypoint( site )
-                                } )
+                            // smk.$tool.directions.active = true
```
Please add a comment indicating why this code is commented out.

dgboss commented on src/smk/tool/search/tool-search-location.js:
```diff
@@ -42,26 +42,26 @@ include.module( 'tool-search.tool-search-location-js', [
-            smk.on( this.id, {
-                'directions': function () {
-                    smk.$tool.directions.active = true
+            // smk.on( this.id, {
```
Same here on the comment front please.

michaelpnelson committed:
Remove nonfunctioning calls to other tools

This removes code in tools that call other tools in a way that is nonfunctional.
